pkg/quay/quay_test.go

```diff
@@ -20,199 +20,461 @@ import (
+		},
+	}
 
-	client := &http.Client{Transport: &http.Transport{}}
-	gock.InterceptClient(client)
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			defer gock.Off()
 
-	quayClient := NewQuayClient(client, "authtoken", "https://quay.io/api/v1")
+			// First page
+			response := Response{Repositories: []Repository{{Name: "test1"}}, NextPage: "next_page_token"}
 
-	// Second page
-	response.Repositories = []Repository{{Name: "test2"}}
-	response.NextPage = "next_page_token2"
-
-	gock.New("https://quay.io/api/v1").
-		MatchHeader("Content-Type", "application/json").
-		MatchHeader("Authorization", "Bearer authtoken").
-		MatchParam("next_page", "next_page_token").
-		Get("/repository").
-		Reply(200).
-		JSON(response)
-
-	// Last page
-	response.Repositories = []Repository{{Name: "test3"}}
-
-	gock.New("https://quay.io/api/v1").
-		MatchHeader("Content-Type", "application/json").
-		MatchHeader("Authorization", "Bearer authtoken").
-		MatchParam("next_page", "next_page_token2").
-		Get("/repository").
-		Reply(200).
-		JSON(response)
-
-	receivedRepos, err := quayClient.GetAllRepositories("test_org")
-
-	if err != nil {
-		t.Errorf("Error getting all repositories, Expected nil, got %v", err)
-	}
-	if len(receivedRepos) != 3 {
-		t.Errorf("Possible pagination error, expected 3 repos, got %d repos", len(receivedRepos))
+			resp := gock.New("https://quay.io/api/v1").
+				MatchHeader("Content-Type", "application/json").
+				MatchHeader("Authorization", "Bearer authtoken").
+				Get("/repository").
+				Reply(tc.statusCode)
+
+			if tc.expectedErr == "failed to unmarshal body" {
+				resp.JSON("{\"next_page\": \"page_token}")
+			} else {
+				resp.JSON(response)
+			}
+
+			client := &http.Client{Transport: &http.Transport{}}
+			gock.InterceptClient(client)
+
+			quayClient := NewQuayClient(client, "authtoken", "https://quay.io/api/v1")
+
+			// Second page
+			response.Repositories = []Repository{{Name: "test2"}}
+			response.NextPage = "next_page_token2"
+
+			gock.New("https://quay.io/api/v1").
+				MatchHeader("Content-Type", "application/json").
+				MatchHeader("Authorization", "Bearer authtoken").
+				MatchParam("next_page", "next_page_token").
+				Get("/repository").
+				Reply(200).
+				JSON(response)
+
+			// Last page
+			response.Repositories = []Repository{{Name: "test3"}}
+
+			gock.New("https://quay.io/api/v1").
+				MatchHeader("Content-Type", "application/json").
+				MatchHeader("Authorization", "Bearer authtoken").
+				MatchParam("next_page", "next_page_token2").
+				Get("/repository").
+				Reply(200).
+				JSON(response)
+
+			receivedRepos, err := quayClient.GetAllRepositories("test_org")
+
+			if tc.expectedErr == "" {
+				if err != nil {
+					t.Errorf("Error getting all repositories, Expected nil, got %v", err)
+				}
+				if len(receivedRepos) != 3 {
+					t.Errorf("Possible pagination error, expected 3 repos, got %d repos", len(receivedRepos))
+				}
```

Comment:
`assert.NilError` and `assert.Condition` could be used here, couldn't they?

Reply:
Updated. PTAL.